src/test/unit_tests/test_nc_online_upgrade_s3_integrations.js

```diff
@@ -0,0 +1,190 @@
+            assert.strictEqual(err.Code, 'InternalError');
+            return true;
+        });
+    }, TEST_TIMEOUT);
+
+    mocha.it('create bucket - host is not blocked for config dir updates', async function() {
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.createBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('put bucket policy - host is blocked for config dir updates - should fail', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await assert.rejects(async () => s3_client.putBucketPolicy({ Bucket: bucket_defaults.name, Policy: JSON.stringify(bucket_policy) }),
+            err => {
+                assert.strictEqual(err.Code, 'InternalError');
+                return true;
+            }
+        );
+    }, TEST_TIMEOUT);
+
+    mocha.it('put bucket policy - host is not blocked for config dir updates', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.putBucketPolicy({ Bucket: bucket_defaults.name, Policy: JSON.stringify(bucket_policy) });
+    }, TEST_TIMEOUT);
+
+    mocha.it('delete bucket - host is blocked for config dir updates - should fail', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await assert.rejects(async () => s3_client.deleteBucket({ Bucket: bucket_defaults.name }), err => {
+            assert.strictEqual(err.Code, 'InternalError');
+            return true;
+        });
+    }, TEST_TIMEOUT);
+
+    mocha.it('delete bucket - host is not blocked for config dir updates', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.deleteBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('head bucket', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await s3_client.headBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('PUT object - host is blocked for config dir updates', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await s3_client.putObject({ Bucket: bucket_defaults.name, Key: mock_obj_key, Body: mock_obj_body });
+    }, TEST_TIMEOUT);
+
+    mocha.it('GET object - host is blocked for config dir updates', async function() {
```

Comment:
may be something like 
`GET object - host is blocked for config dir updates - Should not fail object action`
Will help to understand the test better.